agent/src/main/java/com/microsoft/applicationinsights/agent/internal/agent/CustomClassWriterv2.java

```diff
@@ -0,0 +1,138 @@
+package com.microsoft.applicationinsights.agent.internal.agent;
+
+import org.objectweb.asm.ClassReader;
+import org.objectweb.asm.Opcodes;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+/**
+ * This class overwrites default class writer of ASM to use the ClassLoader
+ * provided by DefaultByteCode transformer (This loader essentially has all
+ * the required classes already loaded)
+ */
+public class CustomClassWriterv2 extends org.objectweb.asm.ClassWriter {
+
+    private ClassLoader classLoader;
+
+    public CustomClassWriterv2(int writerFlag, ClassLoader loader) {
+        super(writerFlag);
+        this.classLoader = loader;
+    }
+
+    /**
+     * This method returns common super class for both the classes without actually loading
+     * the class using forName(). If no super class is present it returns java/lang/Object class.
+     * @param type1
+     * @param type2
+     * @return The String for the common super class of both the classes
+     */
+    @Override
+    protected String getCommonSuperClass(final String type1, final String type2) {
```

Comment:
At the very least this method should have a unit test, if not the whole class.

Reply:
Sure I will get couple of them.